datalake/src/main/java/com/sequenceiq/datalake/service/sdx/stop/SdxStopService.java

```diff
@@ -100,4 +123,18 @@ public void stopAllDatahub(Long sdxId) {
         SdxCluster sdxCluster = sdxService.getById(sdxId);
         distroxService.stopAttachedDistrox(sdxCluster.getEnvCrn());
     }
+
+    public boolean checkIfStoppable(SdxCluster cluster) {
+        Optional<FlowLog> lastFlowLog = flowLogService.getLastFlowLog(cluster.getId());
+        if (lastFlowLog.isPresent()) {
+            String flowChainType = flowChainLogService.getFlowChainType(
+                    lastFlowLog.get().getFlowChainId()
+            );
+            if (flowChainType != null && UNSTOPPABLE_FLOWS.contains(flowChainType)) {
```

Comment:
What happens if the last flow chain was a data lake resize but already completed. Does this logic allow the stop then? It seems like this code won't allow stopping the cluster then. Am I right?

Reply:
I believe I have avoided this scenario by adding an additional check for if the flow log is not finalized. I believe that this will ensure the flow chain type check is for a current active flow chain and not a previously active one.